### ISSUE TITLE
Error: ENOENT: no such file or directory

### ISSUE THREAD

**Issue by berkaey** [marcomontalbano/figma-export#25]
Hey there,

Am I doing something wrong here?

The error;
```
Error: ENOENT: no such file or directory, open '/foo/bar/output/Resources-icons/wish.svg'
    at components.forEach (~/bar/node_modules/@figma-export/output-components-as-svg/index.js:11:20)
    at Array.forEach (<anonymous>)
    at pages.forEach (~/bar/node_modules/@figma-export/output-components-as-svg/index.js:9:24)
    at Array.forEach (<anonymous>)
    at ~/bar/node_modules/@figma-export/output-components-as-svg/index.js:8:15
    at Promise.all.outputters.map (~/bar/node_modules/@figma-export/core/lib/export-components.js:21:53)
    at Array.map (<anonymous>)
    at Object.components (~/bar/node_modules/@figma-export/core/lib/export-components.js:21:34)
```

.figmaexportrc.js
```
module.exports = {
  commands: [
    [
      "components",
      {
        fileId: "XXX",
        onlyFromPages: ["Resources"],
        transformers: [
          require("@figma-export/transform-svg-with-svgo")({
            plugins: [{ removeViewBox: false }, { removeDimensions: true }]
          })
        ],
        outputters: [
          require("@figma-export/output-components-as-svg")({
            output: "./output"
          })
        ]
      }
    ]
  ]
};
```
run command
```
figma-export use-config
```

**Comment by berkaey** [comment]
Hmm, I guess I found the issue,

When you creating files you won't generate folders recursively so when node couldn't find the folders it raises errors. I generate the folders manually and it worked. 

btw, my component names is like "icons/wish" or "icons/color/wish" so plugin needs to be create those folder for us.

and

is there an option to manipulate the svg and the file name before creating the file?

also 

is there any option to remove "PageName-" while creating the parent folder inside output folder?

**Comment by marcomontalbano** [comment]
Hi @berkaey, thanks for this request. What you are asking for is not yet implemented on the library but I'm going to start implementing your suggested features. I created two new tickets to track their development (#26 #27).

I have just a few questions:
1. when you say "option to manipulate the svg before creating the file", which kind of manipulation you need? You can use `@figma-export/transform-svg-with-svgo` to manipulate the generated svg. My plugin uses [SVGO](https://github.com/svg/svgo) behind the scenes, so you can use their options as you need.
1. Instead of adding an option to remove `PageName-` I was thinking about creating a folder with `PageName`. What do you think? For example as your structure you would see something like this:
```
output
└── Resources
    └── icons
        ├── color
        │   └── wish.svg
        └── wish.svg
```
3. when you say "manipulate the filename before creating the file" you need to add prefix/suffix or manipulate the entire name? Can you provide some example?

Thanks a lot! Your requests will help improving `figma-export` for everyone!

**Comment by berkaey** [comment]
I'm sharing the my final `.figmaexportrc` (it is a little bit messy :) ) it can give you some ideas.

My specific use case was to convert resources to `react-native-svg` files.

maybe you can clean this up an give a specific entry point to achieve this use case.

```const fs = require("fs");
const path = require("path");
const svgr = require("@svgr/core");

function toTitleCase(str) {
  return str
    .toLowerCase()
    .replace(/(?:^|\s|\/|\-)\w/g, function(match) {
      return match.toUpperCase();
    })
    .replace(/[^\w\s]/gi, "");
}

const getVariableName = componentName => {
  const variableName = toTitleCase(componentName);

  if (/^[\d]+/.test(variableName)) {
    throw new Error(
      `"${componentName}" - Component names cannot start with a number.`
    );
  }

  return variableName;
};

module.exports = {
  commands: [
    [
      "components",
      {
        fileId: "XXXX",
        onlyFromPages: ["Resources"],
        transformers: [
          require("@figma-export/transform-svg-with-svgo")({
            plugins: [{ removeViewBox: false }, { removeDimensions: true }]
          })
        ],
        outputters: [
          async pages => {
            console.log("");
            const output = "./src/assets";
            fs.mkdirSync(output, { recursive: true });

            let index = {};
            let others = "";

            try {
              for (const { name: pageName, components } of pages) {
                for (const { name: componentName, svg } of components) {
                  const type = componentName.split("/")[0];

                  const varName = getVariableName(
                    componentName.split("/").pop()
                  );

                  let file = "";
                  let replaceAttrValues = {};
                  
                  if (type == "icons" || type == "logo") {
                    replaceAttrValues = {
                      "#6686F6": "{props.fill}",
                      "#1E1739": "{props.fill}",
                      "#000": "{props.fill}"
                    };
                  }

                  file = await svgr.default(
                    svg,
                    {
                      native: true,
                      replaceAttrValues
                    },
                    { varName }
                  );

                  file = file.replace(
                    "export default SvgComponent",
                    `export const ${varName} = SvgComponent`
                  );

                  const filePath = path.resolve(output, `${componentName}.js`);

                  let p = filePath.split("/");
                  p.pop();
                  p = p.join("/");

                  fs.mkdirSync(p, { recursive: true });

                  fs.writeFileSync(filePath, file);

                  if (type) {
                    if (!index[type]) index[type] = "";
                    index[type] = `${
                      index[type]
                    }export * from "./${componentName.split("/").pop()}"\n`;
                  } else {
                    others = `${others}\nexport * from "./${componentName}";`;
                  }
                  if (type) {
                    console.log("GENERATED", varName, "as", type);
                  } else {
                    console.log("GENERATED", varName);
                  }
                }
              }
            } catch (e) {
              console.log(e);
            }

            for (let key of Object.keys(index)) {
              fs.writeFileSync(
                path.resolve(output, `${key}/index.js`),
                index[key]
              );
            }
            if (others.length) {
              fs.writeFileSync(
                path.resolve(output, `${key}/index.js`),
                index[key]
              );
            }
          }
        ]
      }
    ]
  ]
};
```

**Comment by marcomontalbano** [comment]
To reduce complexity in your configuration file you can extract your function into a separate file so you have a `.figmaexportrc` clean and readable.
Regarding your configuration I see few lines that are too specific to your implementation, I'm not sure that we can generalize them. I can write a configurable `svgr` plugin for `figma-export`, trying to achieve your same result (thanks for suggestion).
Anyway writing your own `figma-export` plugin is totally fine.

**Comment by marcomontalbano** [comment]
Hi, the issue you mentioned opening this ticket has been fixed (#26). I also created #34 to track SVGR support.
